### ISSUE TITLE
[Java-PlayFramework] Fix names for the Controller interfaces

### ISSUE THREAD

**Issue by tzimisce012** [swagger-api/swagger-codegen#6099]
The name of the api interfaces are written as "{{classname}}ControllerImpInterfaces" and the implementation as "{{classname}}ControllerImp. 

It would make more sense to call the interfaces as "{{classname}}ControllerInterface" without the word Imp in the middle.

**Comment by JFCote** [comment]
Hi @tzimisce012 , thanks for the suggestion.

Right now, the logic is as follow:

- `{{classname}}Controller` = The controller itself. Should always be generated and overwritten at each generation.

- `{{classname}}ControllerImp` = The actual implementation which contains your business logic. Should only be generated the first time but not after to prevent replacing your business logic.

- `{{classname}}ControllerImpInterface` = The interface that `controllerImp` is following. Should always be generated too.

I would be strange to rename `{{classname}}ControllerImpInterface` without the `imp` because it would look like it's an interface implemented by the controller itself which is not the case at all. But I can understand why some people may not like the fact that `imp` and `interface` end up in the same class name.

Maybe we could rename things like this:

`{{classname}}Controller` -> Stay the same
`{{classname}}ControllerLogic` -> We remove the `imp` from the name for a clearer name and replace it with `Logic`
`{{classname}}ControllerLogicInterface` ->Same thing

At the end of the day, it's only to replace "imp" by "Logic" if that can help people to understand the design better. But we could also keep it the way it is since it doesn't change change anything in the end.

What do you think?
@wing328 : You advice is welcome too!

**Comment by wing328** [comment]
> At the end of the day, it's only to replace "imp" by "Logic"

Other Java server generators use "impl" instead so I would suggest we keep the same convention.

**Comment by tzimisce012** [comment]
ok

**Comment by JFCote** [comment]
@tzimisce012 . you mean adding a "l" after "imp" to get "impl"?

**Comment by tzimisce012** [comment]
@JFCote no, that is irrelevant. You understood me first time. It sounded a bit weird for me to have ImpInterface and something like Logic might sound a bit better. 

But as @wing328 said, this is the notation that all java servers are using so we should keep using them

**Comment by JFCote** [comment]
Ok thanks for the feedback!

**Comment by JFCote** [comment]
@wing328 Please close this issue when you'll have time. Thanks
